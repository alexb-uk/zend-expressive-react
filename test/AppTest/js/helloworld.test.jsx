// __tests__/helloworld.test.jsx

import React from "react";
import renderer from "react-test-renderer";
import Helloworld from "../../../src/App/js/components/helloworld";

test("Helloworld Like counter", () => {
  const component = renderer.create(
    <Helloworld date={new Date("2001/02/03 01:20:30")}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // console.log(tree.children[4].children[0]);

  // manually trigger the like callback
  let link = tree.children[4].children[0];
  link.props.onClick();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // click again
  link.props.onClick();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});
