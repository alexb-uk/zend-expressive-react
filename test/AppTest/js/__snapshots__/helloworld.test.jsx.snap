// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Helloworld Like counter 1`] = `
<div>
  <h1>
    Hello, world!
  </h1>
  <h2>
    It is 
    01:20:30
     precisely!
  </h2>
  Likes : 
  <span>
    0
  </span>
  <div>
    <button
      onClick={[Function]}
    >
      Like Me
    </button>
  </div>
</div>
`;

exports[`Helloworld Like counter 2`] = `
<div>
  <h1>
    Hello, world!
  </h1>
  <h2>
    It is 
    01:20:30
     precisely!
  </h2>
  Likes : 
  <span>
    1
  </span>
  <div>
    <button
      onClick={[Function]}
    >
      Like Me
    </button>
  </div>
</div>
`;

exports[`Helloworld Like counter 3`] = `
<div>
  <h1>
    Hello, world!
  </h1>
  <h2>
    It is 
    01:20:30
     precisely!
  </h2>
  Likes : 
  <span>
    2
  </span>
  <div>
    <button
      onClick={[Function]}
    >
      Like Me
    </button>
  </div>
</div>
`;
